sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/util/ValidationUtil.java

```diff
@@ -64,4 +63,8 @@ public static void validateInteractiveBrowserRedirectUrlSetup(Integer port, Stri
                                                  + " the redirect URL on the builder."));
         }
     }
+
+    private static boolean isValidTenantCharacter(char c) {
+        return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || (c == '.') || (c == '-');
```

Comment:
is `.` actually valid? Or was there just a bug in the regex?

Reply:
yes, it is an accepted character for client ids.